test/tests/functional/pbs_qstat_count.py

```diff
@@ -0,0 +1,287 @@
+        restart.
+        (each job uses ncpus=1)
+        """
+        # create 2 execution queues
+        qname = ['workq1', 'workq2']
+        for i in range(2):
+            a = {
+                'queue_type': 'Execution',
+                'enabled': 'True',
+                'started': 'True'}
+            self.server.manager(MGR_CMD_CREATE, QUEUE,
+                                a, qname[i], expect=True)
+
+        q1_attr = {ATTR_queue: 'workq1'}
+        q2_attr = {ATTR_queue: 'workq2'}
+
+        # submit 1 job per queue to ensure a running job in each queue,
+        # then submit 2 more jobs per queue i.e. overall 3 jobs in each queue
+        j = Job(TEST_USER, q1_attr)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        j = Job(TEST_USER, q2_attr)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        for i in range(2):
+            j = Job(TEST_USER, q1_attr)
+            self.server.submit(j)
+            j = Job(TEST_USER, q2_attr)
+            self.server.submit(j)
+
+        # no. of R jobs = 2, as available ncpus=2
+        self.verify_count(2)
+
+    def test_queued_restart_multiple_queue(self):
+        """
+        The test case verifies that the queued_count reported in the output
+        of qstat -Bf is equal to total_jobs - running jobs, even after the
+        server is restart.
+        (each job uses ncpus=1)
+        """
+        # create 2 execution queues
+        qname = ['workq1', 'workq2']
+        for i in range(2):
+            a = {
+                'queue_type': 'Execution',
+                'enabled': 'True',
+                'started': 'True'}
+            self.server.manager(MGR_CMD_CREATE, QUEUE,
+                                a, qname[i], expect=True)
+
+        q1_attr = {ATTR_queue: 'workq1'}
+        q2_attr = {ATTR_queue: 'workq2'}
+
+        # submit 1 job per queue to ensure a running job in each queue,
+        # then submit 2 more jobs per queue i.e. overall 3 jobs in each queue
+        j = Job(TEST_USER, q1_attr)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        j = Job(TEST_USER, q2_attr)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        for i in range(2):
+            j = Job(TEST_USER, q1_attr)
+            self.server.submit(j)
+            j = Job(TEST_USER, q2_attr)
+            self.server.submit(j)
+
+        self.server.restart()
+        # no. of R jobs = 2, as available ncpus=2
+        self.verify_count(2)
+
+    def test_queued_sched_false(self):
+        """
+        This test case verifies that the value of queued_count in the output
+        of qstat -Bf matches the no. of jobs submitted (each using ncpus=1),
+        as scheduling is set to False.
+        """
+        # set scheduling false
```

Comment:
Comments should explain why, not what.  The line below this says the same thing.  Either explain why you're doing this or remove the comment.